kura/org.eclipse.kura.driver.ble.sensortag.provider/src/main/java/org/eclipse/kura/internal/driver/ble/sensortag/TiSensorTag.java

```diff
@@ -0,0 +1,1503 @@
+        BluetoothLeGattCharacteristic magValueChar;
+        try {
+            if (this.cc2650) {
+                magValueChar = this.gattServices.get(MOVEMENT)
+                        .findCharacteristic(TiSensorTagGatt.UUID_MOV_SENSOR_VALUE);
+                magValueChar.disableValueNotifications();
+            } else {
+                magValueChar = this.gattServices.get(MAGNETOMETER)
+                        .findCharacteristic(TiSensorTagGatt.UUID_MAG_SENSOR_VALUE);
+                magValueChar.disableValueNotifications();
+            }
+        } catch (KuraException e) {
+            logger.error("Magnetic field notification disable failed", e);
+        }
+    }
+
+    /*
+     * Set sampling period
+     */
+    public void setMagnetometerPeriod(int period) {
+        byte[] periodBytes = { ByteBuffer.allocate(4).putInt(period).array()[3] };
+        BluetoothLeGattCharacteristic magPeriodChar;
+        try {
+            if (this.isCC2650()) {
+                magPeriodChar = this.gattServices.get(MOVEMENT)
+                        .findCharacteristic(TiSensorTagGatt.UUID_MOV_SENSOR_PERIOD);
+                magPeriodChar.writeValue(periodBytes);
+            } else {
+                magPeriodChar = this.gattServices.get(MAGNETOMETER)
+                        .findCharacteristic(TiSensorTagGatt.UUID_MAG_SENSOR_PERIOD);
+                magPeriodChar.writeValue(periodBytes);
+            }
+        } catch (KuraException e) {
+            logger.error("Magnetometer period set failed", e);
+        }
+    }
+
+    /*
+     * Calculate Magnetic Field
+     */
+    private float[] calculateMagneticField(byte[] valueByte) {
+
+        logger.info("Received magnetometer value: " + byteArrayToHexString(valueByte));
+
+        float[] magneticField = new float[3];
+        if (this.cc2650) {
+
+            final float scale = (float) 32768 / 4912;
+
+            int x = shortSignedAtOffset(valueByte, 12);
+            int y = shortSignedAtOffset(valueByte, 14);
+            int z = shortSignedAtOffset(valueByte, 16);
+
+            magneticField[0] = x / scale;
+            magneticField[1] = y / scale;
+            magneticField[2] = z / scale;
+        } else {
+
+            int x = shortSignedAtOffset(valueByte, 0);
+            int y = shortSignedAtOffset(valueByte, 2);
+            int z = shortSignedAtOffset(valueByte, 4);
+
+            magneticField[0] = x * (2000f / 65536f) * -1;
+            magneticField[1] = y * (2000f / 65536f) * -1;
+            magneticField[2] = z * (2000f / 65536f);
+        }
+
+        return magneticField;
+    }
+
+    // ------------------------------------------------------------------------------------------------------------------
+    //
+    // Barometric Pressure Sensor
+    //
+    // ------------------------------------------------------------------------------------------------------------------
+
+    /*
+     * Enable pressure sensor
+     */
+    public void enableBarometer() {
```

Comment:
All enableX and disableX in this class could return boolean for better understanding of the API and their respective invocations.

Reply:
Generally speaking, I agree with you. However, the underlying native lib doesn't return any result from write operations. If something goes wrong, it launches an exception.

Reply:
Native libraries are always difficult to manage 😉. You can catch this exception and return accordingly.